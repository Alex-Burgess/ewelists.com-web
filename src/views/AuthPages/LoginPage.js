import React, { useState, useEffect } from 'react';
import { Link } from "react-router-dom";
import { Auth } from "aws-amplify";
import qs from "qs";
// libs
import { onAuthError } from "libs/errorLib";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// components
import FooterGrey from "components/Footer/FooterGrey.js";
import HeaderWhite from "components/Header/HeaderWhite.js";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/Buttons/Button.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import Input from "components/Input/CustomInput.js";
import ErrorText from "components/Typography/Error.js";
// sections
import SocialButtons from "./Sections/SocialButtons.js";

import styles from "assets/jss/material-kit-pro-react/views/authPageStyle.js";
const useStyles = makeStyles(styles);

export default function LoginPage(props) {
  const classes = useStyles();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');

  useEffect( () => {
    function checkUrlParams() {
      const params = qs.parse(props.location.search, { ignoreQueryPrefix: true });

      if (params['error']) {
        switch (params['error']) {
          case "GoogleDomainError":
            setError('An account has already been registed with this email address.  Log in with your username and password.');
            break;
          default:
            setError('Oops, there was error whilst logging in. If this issue persists please contact us.');
            break;
        }
      }
    };

    checkUrlParams();
  }, [props.location.search]);

  const validateForm = () => {
    return email.length > 0 && password.length > 0;
  }

  const handleSubmit = async event => {
    event.preventDefault();

    try {
      await Auth.signIn(email, password);
      props.userHasAuthenticated(true);
    } catch (e) {
      onAuthError(e, email);

      if (e.message === 'User does not exist.') {
        setError("There is no account with the email provided.");
      } else if (e.message === 'Incorrect username or password.') {
        setError("The email or password you provided was incorrect.");
      } else {
        setError(e.message);
      }
    }
  }

  return (
    <div className={classes.page}>
      <HeaderWhite isAuthenticated={false} mobile={props.mobile} tablet={props.tablet}/>
      <div className={classes.container}>
        <GridContainer justify="center">
          <GridItem xs={12} sm={10} md={6} className={classes.gridLogin}>
            <Card className={classes.cardLogin} data-cy="card">
              <CardBody signup>
                <h3 className={classes.title + " " + classes.textCenter}>Log in</h3>
                <SocialButtons />
                <form onSubmit={handleSubmit}>
                  <Input
                    labelText="Email"
                    id="email"
                    formControlProps={{
                      fullWidth: true,
                      value: email,
                      onChange: event => setEmail(event.target.value),
                      className: classes.inputProps
                    }}
                  />
                  <Input
                    labelText="Password"
                    id="password"
                    formControlProps={{
                      fullWidth: true,
                      value: email,
                      onChange: event => setPassword(event.target.value),
                      className: classes.loginInputs
                    }}
                    inputProps={{
                      type: "password",
                      autoComplete: "off"
                    }}
                  />
                  { error
                    ? <div className={classes.textCenter}>
                        <ErrorText>
                          <p>{error}</p>
                        </ErrorText>
                      </div>
                    : null
                  }
                  <Button fullWidth color="primary" type="submit" data-cy="login" disabled={!validateForm()}>
                    Login
                  </Button>
                  <Link to="/reset" className={classes.link}>Forgot your password?</Link>
                </form>
                <div className={classes.accountCheck + " " + classes.textCenter}>
                  Don't have an account? <Link to="/signup" data-cy="link-signup">Sign Up</Link>
                </div>
              </CardBody>
            </Card>
          </GridItem>
        </GridContainer>
      </div>
      <div className={classes.flexer} />
      <FooterGrey />
    </div>
  );
}
