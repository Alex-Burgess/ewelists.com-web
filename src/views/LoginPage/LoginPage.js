import React, { useState } from 'react';
import { Auth } from "aws-amplify";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import InputAdornment from "@material-ui/core/InputAdornment";
import Icon from "@material-ui/core/Icon";
// @material-ui/icons
import Email from "@material-ui/icons/Email";
// core components
import FooterTransparent from "custom/Footer/FooterTransparent.js";
import HeaderTransparent from "custom/Header/HeaderTransparent.js";
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Button from "components/CustomButtons/Button.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import CardHeader from "components/Card/CardHeader.js";
import CustomInput from "components/CustomInput/CustomInput.js";
import config from 'config.js';

import styles from "assets/jss/custom/views/loginPageStyle.js";
const useStyles = makeStyles(styles);

const backgroundImage = config.imagePrefix + "/images/sheep-with-shoes.jpg";

export default function LoginPage(props) {
  const classes = useStyles();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');

  const validateForm = () => {
    return email.length > 0 && password.length > 0;
  }

  const checkGoogleEmail = (email) => {
    if (email.includes('@googlemail.com')) {
      var fields = email.split('@');
      email = fields[0] + "@gmail.com"
    }
    return email
  }

  const handleSubmit = async event => {
    event.preventDefault();

    let checkedEmail = checkGoogleEmail(email)

    try {
      await Auth.signIn(checkedEmail, password);
      props.userHasAuthenticated(true);
    } catch (e) {
      setError(e.message);
    }
  }

  return (
    <div>
      <HeaderTransparent isAuthenticated={false} />
      <div
        className={classes.pageHeader}
        style={{
          backgroundImage: "url(" + backgroundImage + ")",
          backgroundSize: "cover",
          backgroundPosition: "top center"
        }}
      >
        <div className={classes.container}>
          <GridContainer justify="center">
            <GridItem xs={12} sm={10} md={6}>
              <Card className={classes.cardLogin}>
                  <CardHeader
                    color="info"
                    signup
                    className={classes.cardHeader}
                  >
                    <h4 className={classes.cardTitle}>Login</h4>
                    <div className={classes.socialLine}>
                      <Button
                        justIcon
                        color="transparent"
                        className={classes.iconButtons}
                        onClick={() => Auth.federatedSignIn({provider: 'LoginWithAmazon'})}
                      >
                        <i className="fab fa-amazon" />
                      </Button>
                      <Button
                        justIcon
                        color="transparent"
                        className={classes.iconButtons}
                        onClick={() => Auth.federatedSignIn({provider: 'Google'})}
                      >
                        <i className="fab fa-google" />
                      </Button>
                      <Button
                        justIcon
                        color="transparent"
                        className={classes.iconButtons}
                        onClick={() => Auth.federatedSignIn({provider: 'Facebook'})}
                      >
                        <i className="fab fa-facebook" />
                      </Button>
                    </div>
                  </CardHeader>
                  <p
                    className={classes.description + " " + classes.textCenter}
                  >
                    Or log in with email
                  </p>
                  <CardBody signup>
                    <form className={classes.form} onSubmit={handleSubmit}>
                    <CustomInput
                      id="email"
                      formControlProps={{
                        fullWidth: true,
                        value: email,
                        onChange: event => setEmail(event.target.value)
                      }}
                      inputProps={{
                        placeholder: "Email...",
                        type: "email",
                        startAdornment: (
                          <InputAdornment position="start">
                            <Email className={classes.inputIconsColor} />
                          </InputAdornment>
                        )
                      }}
                    />
                    <CustomInput
                      id="password"
                      formControlProps={{
                        fullWidth: true,
                        value: email,
                        onChange: event => setPassword(event.target.value)
                      }}
                      inputProps={{
                        placeholder: "Password",
                        type: "password",
                        startAdornment: (
                          <InputAdornment position="start">
                            <Icon className={classes.inputIconsColor}>
                              lock_utline
                            </Icon>
                          </InputAdornment>
                        ),
                        autoComplete: "off"
                      }}
                    />
                    <div className={classes.details}>
                      <a href="/login/reset" className={classes.link}>Forgot your password?</a>
                    </div>
                    { error
                      ? <div id="loginError" className={classes.loginError}>
                          <p>{error}</p>
                        </div>
                      : null
                    }
                    <div className={classes.textCenter}>
                      <Button round color="info" type="submit" disabled={!validateForm()}>
                        Login
                      </Button>
                    </div>
                  </form>
                  </CardBody>
              </Card>
            </GridItem>
          </GridContainer>
        </div>
        <FooterTransparent />
      </div>
    </div>
  );
}
