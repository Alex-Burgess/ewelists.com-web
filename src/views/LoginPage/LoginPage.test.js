import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import {mount} from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Auth } from "aws-amplify";
import LoginPage from './LoginPage';

// TODO - Test username and password login successful.  Tricky to do, because redirects.
// Test currently fails because can't mock props.userHasAuthenticated.

// TODO - Test social login buttons.


describe('Login Page Snapshots', () => {
  test('Renders empty form of login page', () => {
    const tree = TestRenderer
      .create(<Router><LoginPage /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('Test Input Form', () => {
  const wrapper = mount(
    <Router>
      <LoginPage />
    </Router>
  );

  it('Should have 1 form', () => {
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('Should have 2 inputs', () => {
    expect(wrapper.find('CustomInput').length).toEqual(2);
  });

  it('Should have email input', () => {
    expect(wrapper.find('CustomInput[id="email"]').length).toEqual(1);
  });

  it('Should have message input', () => {
    expect(wrapper.find('CustomInput[id="password"]').length).toEqual(1);
  });

  it('Should have 1 form submit button', () => {
    expect(wrapper.find('form').find('button').length).toEqual(1);
  });

  it('Form submit button should be disabled', () => {
    const submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);
  });

  it('Submit button should be active when all fields populated', () => {
    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.update();
    let submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(true);

    wrapper.find('input[id="password"]').simulate('change', {
      target: {
        value: '12345678',
      },
    });

    wrapper.update();
    submitButton = wrapper.find('form').find('button');
    expect(submitButton.hasClass(/(makeStyles-disabled)-(\d+)/)).toEqual(false);
  });
});


describe('Test Username and Password Login', () => {
  const wrapper = mount(
    <Router>
      <LoginPage />
    </Router>
  );

  it('Login should fail with user that does not exist', () => {
    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.find('input[id="password"]').simulate('change', {
      target: {
        value: '12345678',
      },
    });

    wrapper.update();

    Auth.signIn = jest.fn().mockImplementation(() => {
      throw new Error("User does not exist.");
    });

    let form = wrapper.find('form');
    form.simulate('submit');

    expect(wrapper.find('#loginError').text()).toEqual('User does not exist.');
  });

  it('Login should fail with incorrect password', () => {
    wrapper.find('input[id="email"]').simulate('change', {
      target: {
        value: 'test.user@gmail.com',
      },
    });

    wrapper.find('input[id="password"]').simulate('change', {
      target: {
        value: '12345678',
      },
    });

    wrapper.update();

    Auth.signIn = jest.fn().mockImplementation(() => {
      throw new Error("Incorrect username or password.");
    });

    let form = wrapper.find('form');
    form.simulate('submit');

    expect(wrapper.find('#loginError').text()).toEqual('Incorrect username or password.');
  });

  // TODO
  // it('should render landing page after login', async () => {
  //   wrapper.find('input[id="email"]').simulate('change', {
  //     target: {
  //       value: 'test.user@gmail.com',
  //     },
  //   });
  //
  //   wrapper.find('input[id="password"]').simulate('change', {
  //     target: {
  //       value: '12345678',
  //     },
  //   });
  //
  //   wrapper.update();
  //
  //   Auth.signIn = jest.fn().mockImplementation(() => {
  //     return true
  //   });
  //
  //   userHasAuthenticated = jest.fn().mockImplementation(() => {
  //     return true
  //   });
  //
  //   let form = wrapper.find('form');
  //   await act(async () => {
  //     form.simulate('submit');
  //   });
  //
  //   // console.log("Wrapper: " + wrapper.debug())
  //   wrapper.update();
  //   expect(wrapper.find('#loginError').text()).toEqual('Incorrect username or password.');
  // });
});

describe('Test forgot your password link', () => {
  const wrapper = mount(
    <Router>
      <LoginPage />
    </Router>
  );

  it('Should have reset password link', () => {
    expect(wrapper.find('form').find('a').length).toEqual(1);
  });

  it('Should have correct reset password path', () => {
    expect(wrapper.find('form').find('a').props().href).toBe('/login/reset');
  });
});
