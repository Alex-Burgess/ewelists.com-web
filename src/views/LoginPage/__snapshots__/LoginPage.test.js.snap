// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Snapshots Renders empty form of login page 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-41 makeStyles-transparent-53 makeStyles-fixed-43 MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular makeStyles-container-44 MuiToolbar-gutters"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-45"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <img
                alt="logo"
                className="makeStyles-logo-40"
                src="logo-white.png"
              />
            </a>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <div
          className="makeStyles-hidden-57 PrivateHiddenCss-smDown-139"
        >
          <div
            className="makeStyles-collapse-58"
          >
            <ul
              className="MuiList-root makeStyles-list-149 makeStyles-mlAuto-165 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-listItem-150 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-152"
                  href="/list-ideas"
                  onClick={[Function]}
                >
                  Gift List Ideas
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-150 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-152"
                  href="/login"
                  onClick={[Function]}
                >
                  Log In
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-listItem-150 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-navLink-152"
                  href="/signup"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="makeStyles-pageHeader-4"
    style={
      Object {
        "backgroundImage": "url(sheep-with-shoes.jpg)",
        "backgroundPosition": "top center",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="makeStyles-container-3"
    >
      <div
        className="MuiGrid-root makeStyles-grid-181  MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root makeStyles-grid-285  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-6"
        >
          <div
            className="makeStyles-card-286 makeStyles-cardLogin-19"
          >
            <div
              className="makeStyles-cardHeader-301 makeStyles-infoCardHeader-310 makeStyles-cardHeaderSignup-306 makeStyles-cardHeader-6"
            >
              <h4
                className="makeStyles-cardTitle-2"
              >
                Login
              </h4>
              <div
                className="makeStyles-socialLine-7"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-313 makeStyles-transparent-336 makeStyles-justIcon-343 makeStyles-iconButtons-10"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-amazon"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-313 makeStyles-transparent-336 makeStyles-justIcon-343 makeStyles-iconButtons-10"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-google"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-313 makeStyles-transparent-336 makeStyles-justIcon-343 makeStyles-iconButtons-10"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-facebook"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <p
              className="makeStyles-description-1 makeStyles-textCenter-9"
            >
              Or log in with email
            </p>
            <div
              className="makeStyles-cardBody-345 makeStyles-cardSignup-350"
            >
              <form
                className="makeStyles-form-5"
                onSubmit={[Function]}
              >
                <div
                  className="MuiFormControl-root makeStyles-formControl-360 MuiFormControl-fullWidth"
                  onChange={[Function]}
                  value=""
                >
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-353 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                    onClick={[Function]}
                  >
                    <div
                      className="MuiInputAdornment-root MuiInputAdornment-positionStart"
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root makeStyles-inputIconsColor-8"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
                        />
                      </svg>
                    </div>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input MuiInput-input makeStyles-input-362 MuiInputBase-inputAdornedStart"
                      disabled={false}
                      id="email"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Email..."
                      required={false}
                      type="email"
                    />
                  </div>
                </div>
                <div
                  className="MuiFormControl-root makeStyles-formControl-360 MuiFormControl-fullWidth"
                  onChange={[Function]}
                  value=""
                >
                  <div
                    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-353 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
                    onClick={[Function]}
                  >
                    <div
                      className="MuiInputAdornment-root MuiInputAdornment-positionStart"
                    >
                      <span
                        aria-hidden={true}
                        className="material-icons MuiIcon-root makeStyles-inputIconsColor-8"
                      >
                        lock_utline
                      </span>
                    </div>
                    <input
                      aria-invalid={false}
                      autoComplete="off"
                      className="MuiInputBase-input MuiInput-input makeStyles-input-362 MuiInputBase-inputAdornedStart"
                      disabled={false}
                      id="password"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Password"
                      required={false}
                      type="password"
                    />
                  </div>
                </div>
                <div>
                  <a
                    className="makeStyles-link-20"
                    href="/login/reset"
                  >
                    Forgot your password?
                  </a>
                </div>
                <div
                  className="makeStyles-textCenter-9"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-313 makeStyles-info-317 makeStyles-round-340 makeStyles-disabled-337"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="submit"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Login
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      className="makeStyles-footer-439 makeStyles-footer-433"
    >
      <div
        className="makeStyles-container-445"
      >
         
        <div>
          <div
            className="makeStyles-left-430"
          >
            <ul
              className="MuiList-root makeStyles-list-429 MuiList-padding"
            >
              <li
                className="MuiListItem-root makeStyles-inlineBlock-428 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-427"
                  href="/"
                >
                  © 
                  2020
                   ewelists
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-428 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-427"
                  href="/contact"
                >
                  Contact Us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-428 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-427"
                  href="/#about"
                >
                  About us
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-428 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-427"
                  href="/privacy"
                >
                  Privacy
                </a>
              </li>
              <li
                className="MuiListItem-root makeStyles-inlineBlock-428 MuiListItem-gutters"
                disabled={false}
              >
                <a
                  className="makeStyles-block-427"
                  href="/terms"
                >
                  Terms
                </a>
              </li>
            </ul>
          </div>
          <div
            className="makeStyles-rightLinks-435"
          >
            <ul>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-313 makeStyles-simple-335 makeStyles-justIcon-343"
                  href="https://www.facebook.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-facebook"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-313 makeStyles-simple-335 makeStyles-justIcon-343"
                  href="https://twitter.com/ewelists"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-twitter"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
              <li>
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-313 makeStyles-simple-335 makeStyles-justIcon-343"
                  href="https://www.instagram.com/ewelists/"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  target="_blank"
                >
                  <span
                    className="MuiButton-label"
                  >
                    <i
                      className="fab fa-instagram"
                    />
                  </span>
                  <span
                    className="MuiTouchRipple-root"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="makeStyles-clearFix-455"
        />
      </div>
    </footer>
  </div>
</div>
`;
